src/main/java/com/example/gradecalculator/controller/GradesController.java

```diff
@@ -0,0 +1,75 @@
+package com.example.gradecalculator.controller;
+
+import com.example.gradecalculator.enums.GradeTypes;
+import com.example.gradecalculator.mapper.GradeMapper;
+import com.example.gradecalculator.model.GradeTO;
+import com.example.gradecalculator.repository.UserGradeRepository;
+import com.example.gradecalculator.service.GradeService;
+import com.example.gradecalculator.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@RestController
+@RequestMapping("/grades")
+public class GradesController {
+
+    private final GradeService gradeService;
+
+    private final UserService userService;
+    private final UserGradeRepository userGradeRepository;
+    private final GradeMapper gradeMapper;
+
+    @Autowired
+    public GradesController(GradeService gradeService, UserService userService, UserGradeRepository userGradeRepository, GradeMapper gradeMapper) {
+        this.gradeService = gradeService;
+        this.userService = userService;
+        this.userGradeRepository = userGradeRepository;
+        this.gradeMapper = gradeMapper;
+    }
+
+    @PostMapping("/save-grade")
+    public ResponseEntity<?> saveGrade(@RequestBody GradeTO gradeTO, Authentication authentication) {
+        try {
+            var userId = userService.getAuthenticatedUserId(authentication);
+            gradeService.saveGrade(gradeTO, userId);
+            return ResponseEntity.ok().build();
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Error saving grade");
+        }
+    }
+
+
+    @GetMapping("/userGrades")
+    public ResponseEntity<Map<GradeTypes, List<GradeTO>>> showGrades(Authentication authentication) {
+        try {
+            var userId = userService.getAuthenticatedUserId(authentication);
+            Map<GradeTypes, List<GradeTO>> gradeTOs = StreamSupport.stream(userGradeRepository.findByUserId(userId).spliterator(), false)
```

Comment:
as suggested by IntellJ: 
use <method>.stream() instead of wrapping with StreamSupport
```suggestion
            Map<GradeTypes, List<GradeTO>> gradeTOs = userGradeRepository.findByUserId(userId).stream()
```

FYI: please look at IDE suggestions because they often will present you with best practices or better readable code